libcontainer/nsenter/nsexec.c

```diff
@@ -46,6 +49,9 @@ enum sync_t {
 /* JSON buffer. */
 #define JSON_MAX 4096
 
+/* buffer for synchronisation value */
+#define SYNC_BUF_LEN 10
```

Comment:
Since `SYNC_PRIVATE_CGNS` is a single byte, can you just follow the example of all of the other synchronisation code and not use a buffer (use a single character local variable).